src/main/kotlin/app/revanced/patcher/apk/Archive.kt

```diff
@@ -0,0 +1,65 @@
+package app.revanced.patcher.apk
+
+import app.revanced.patcher.util.xml.LazyXMLInputSource
+import com.reandroid.apk.ApkModule
+import com.reandroid.archive.ByteInputSource
+import com.reandroid.archive.InputSource
+import com.reandroid.arsc.chunk.xml.ResXmlDocument
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.XMLException
+import java.io.IOException
+
+internal data class FileHandle(val virtualPath: String, val archivePath: String, val callback: (() -> Unit)? = null)
```

Comment:
What you did is not wrong, but usually, it's done like this instead:
```suggestion
internal data class FileHandle(val virtualPath: String, val archivePath: String, val callback: () -> Unit = {})
```

Reply:
I did not know empty lambdas are preferred in kotlin. Thanks